src/main/java/org/sagebionetworks/bridge/spring/controllers/HealthDataController.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * API to allow consented users to submit health data in a synchronous API, instead of using the asynchronous
+     * upload API. This is most beneficial for small data sets, like simple surveys. This API returns the health data
+     * record produced from this submission, which includes the record ID.
+     */
+    @PostMapping("/v3/healthdata")
+    @ResponseStatus(HttpStatus.CREATED)
+    public String submitHealthData() throws IOException, UploadValidationException {
+        // Submit health data.
+        UserSession session = getAuthenticatedAndConsentedSession();
+        HealthDataSubmission healthDataSubmission = parseJson(HealthDataSubmission.class);
+        HealthDataRecord savedRecord = healthDataService.submitHealthData(session.getStudyIdentifier(),
+                session.getParticipant(), healthDataSubmission);
+
+        // Write record ID into the metrics, for logging and diagnostics.
+        Metrics metrics = getMetrics();
+        if (metrics != null) {
+            metrics.setRecordId(savedRecord.getId());
+        }
+
+        // Record upload time to user's request info. This allows us to track the last time the user submitted.
+        RequestInfo requestInfo = getRequestInfoBuilder(session).withUploadedOn(DateUtils.getCurrentDateTime())
+                .build();
+        cacheProvider.updateRequestInfo(requestInfo);
+
+        // Return the record produced by this submission. Filter out Health Code, of course.
+        return HealthDataRecord.PUBLIC_RECORD_WRITER.writeValueAsString(savedRecord);
+    }
+
+    /** Allows a developer to submit health data on behalf of the participant. This is generally used for backfills. */
+    @PostMapping("/v3/participants/{userId}/healthdata")
+    @ResponseStatus(HttpStatus.CREATED)
+    public String submitHealthDataForParticipant(@PathVariable String userId) throws IOException, UploadValidationException {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        Study study = studyService.getStudy(session.getStudyIdentifier());
+
+        // Get participant.
+        StudyParticipant participant = participantService.getParticipant(study, userId, false);
+
+        // Submit health data.
+        HealthDataSubmission healthDataSubmission = parseJson(HealthDataSubmission.class);
+        HealthDataRecord savedRecord = healthDataService.submitHealthData(study.getStudyIdentifier(), participant,
+                healthDataSubmission);
+
+        // Write record ID into the metrics, for logging and diagnostics.
+        Metrics metrics = getMetrics();
+        if (metrics != null) {
+            metrics.setRecordId(savedRecord.getId());
+        }
+
+        // Return the record produced by this submission. Filter out Health Code, of course.
+        return HealthDataRecord.PUBLIC_RECORD_WRITER.writeValueAsString(savedRecord);
+    }
+
+    // Not mapped in the BridgePF routes file.
+    /* public StatusMessage updateRecordsStatus() {
```

Comment:
This is used by /v3/recordexportstatuses and /v3/recordExportStatuses (Note there was a capitalization snafu when this was implemented. I'm not sure which one is actually in use right now.) This is used by the Bridge Exporter.

Similarly, we should also add the unit test back.

Reply:
Thanks for finding this... added back along with test.